src/Cache/Engine/RedisEngine.php

```diff
@@ -269,9 +269,8 @@ public function add($key, $value)
             $value = serialize($value);
         }
 
-        // setNx() doesn't have an expiry option, so follow up with an expiry
-        if ($this->_Redis->setNx($key, $value)) {
-            return $this->_Redis->expire($key, $duration);
+        if ($this->_Redis->set($key, $value, ['nx', 'ex' => $duration])) {
```

Comment:
What will happen for older versions of redis?

Reply:
@ethanpooley Ping

Reply:
Gah, sorry. Yes. I have no idea what would happen for older versions. And perhaps I need to stop proposing things if I'm not going to stick with them! Are we checking versions anywhere else, and treating different ones differently, within the Redis engine? If so we could simply version this feature and use the old method for pre-2.6.12.

Alternatively, it seems we should be aware of a minimum version number of Redis that we actually intend to support, and support elsewhere in the engine. 2.6.12 was [released six and a half years ago](https://github.com/antirez/redis/releases/tag/2.6.12), and the version of 2.x that seemed to have much of a legacy life beyond that point was 2.8. Do we have an explicit version support target for our Redis engine?

Reply:
No specifically. This is the first time we have added functionality that depends on 'new' behavior in redis. It looks like ubuntu 18.04 comes with [redis 4.6](https://launchpad.net/ubuntu/bionic/+source/redis) which should make this relatively safe.